### DISCUSSION TITLE
UART MOD FOR 4.2.2

### DISCUSSION THREAD

**Discussion by Boomklimmer** [mriscoc/Ender3V2S1#1321] [Ideas]
Hi Guys,
I want to do the UART Conversion based on this Form:https://github.com/mriscoc/Ender3V2S1/discussions/1259.

I want to do this because I have a BMG Clone and the TMC2208 Just shuts down with Linear advance. Too many steps, too quickly. 

I am currently using this config 

Printer:ender3v2
BOARD:422
LEVELING: BLT, UBL
DISPLAY:DWIN
FEATURES
-CVLM
-HOMEOFFS
-IS
-LA
-MPC

Unfortunately, I have a C6 and not a T6 processor when compiling all of this, I get an error: _region `FLASH' overflowed by 8380 bytes_
Space is all taken up what can I let go for this UART mod?

I have disabled 
-MPC
-The filament ran out but what else can be thrown away Besides IS and LA?

**Comment by mriscoc** [comment]
Check: https://github.com/mriscoc/Ender3V2S1/issues/648#issuecomment-1382675058, https://github.com/mriscoc/Ender3V2S1/issues/502#issuecomment-1318067734

**Comment by Boomklimmer** [comment]
@mriscoc thank you for the reply, I decided to go with the Big Tree tech SKR mini V3 as it has UART and better stepper motor drivers out of the BOX. I tried to use the special configurator:
![image](https://github.com/mriscoc/Ender3V2S1/assets/47988076/fde2ed4d-803d-4584-8127-6ba9abee7228)

And I am struggling to get the BLtouch working. I have tried the 5-pin and the 3+2-pin methods but with no success. The blue light on the BLTOUCH comes on for a fraction of a second but then disappears. 

I have downloaded this preconfigured file and the issue is solved:
![image](https://github.com/mriscoc/Ender3V2S1/assets/47988076/a6894b62-345e-4ac4-b3c0-647b636fd5a9)


 would it be possible to share the Configurations for this so that I can see what the difference is and change the configuration for the SKR MINI V3 to help solve this issue? I will send the Config Files back to you with the change after testing has been completed. I want to add LS and Pressure advance to my build
 
 Thank you for all the hard work on this firmware

**Reply by mriscoc** [reply]
Hi, you should get the same result if you just select Ender 3V2, SKRM3V3, BLT, UBL, DWIN, T1 and MPC in the Configurator.

**Reply by Boomklimmer** [reply]
OK, let me try and work back from there :)

I will post my findings

**Comment by Boomklimmer** [comment]
Hi to anyone using the SKR MINI E V3 Don't fall into the trap of selecting CVLM
![image](https://github.com/mriscoc/Ender3V2S1/assets/47988076/767d704c-0d95-4f16-93e7-45f8f2113fa8).

Thanks @mriscoc for sending me on the correct path.
